### ISSUE TITLE
[Bug]: Velocity is wrong at 10 fps

### ISSUE THREAD

**Issue by Niterux** [TreyRuffy/BetterF3#160]
### Describe the Bug

When the FPS is limited to 10, velocity is reported double what it truly is

### Steps to Reproduce

- Limit FPS to 10
- Walk in a straight line
- Observe "Horizontal Velocity: 8.634"

### Expected Behavior

Velocity being reported as the correct value at lower framerates

### Screenshots

<img width="854" height="480" alt="Image" src="https://github.com/user-attachments/assets/89a964d3-7e5b-418c-8704-7faab5763af2" />

### Modloader Type

Fabric

### Modloader Version

0.17.2

### Mod Version

15.0.0

### Log Output

```shell
I cannot because "Field can not be longer than 65536 characters." but nothing is relevant anyways.
```
